GVRf/Framework/src/org/gearvrf/GVRHybridObject.java

```diff
@@ -310,7 +346,9 @@ private void close() {
                         handler.nativeCleanup(mNativePointer);
                     }
                 }
-                NativeHybridObject.delete(mNativePointer);
+                if (mOwnNative) {
```

Comment:
The whole purpose of the GVRHybridObject is to manage the native peer lifecycle automatically. Now it may or may not perform its very purpose. If a class doesn't need automatic native peer tracking then it shouldn't be deriving from GVRHybridObject.